### DISCUSSION TITLE
BaramMesh stuck on base grid

### DISCUSSION THREAD

**Discussion by samc99us** [nextfoam/baram#92] [General]
First I wanted to say thank you for developing this software, so far it more user friendly than other Openfoam GUI's I have used in the past. Unfortunately I ran into an error while generating a mesh for my specific test case (see attached), BaramMesh locks up during base grid generation, while "Decomposing Case". If I force windows to close BaramMesh then re-load the case file, a BlockMesh is generated and I can move onto the next steps. I think this might have to do with the number of parallel processors selected; dropping from 16 (I have 24 available) to 8 fixed the issue. Is there a max number of parallel processors allowed?

**Comment by kdg5424** [comment]
Hello @samc99us this is Donggyu from NEXTFOAM.

Thanks for your interest in BARAM.
First of all, I want to check that Is it right what you mean "**BaramMesh locks up** is the below picture?
![image](https://github.com/user-attachments/assets/d70eaff1-d1a1-4b06-ad45-2593b3bd77ec)

Then, it is not the issue with the number of parallel processors.
It locks the mesh so that it is pemanently confirmed.

There is no limitation about parallel processor core number and of coure you can change your core number in settings while generating the mesh.

![image](https://github.com/user-attachments/assets/bd861539-b0ed-464d-ac6a-54a7d9c1a8a4)

I wish the reason of the problem is right.
If it is not, let me know again.

Thanks

Donggyu kim

**Comment by samc99us** [comment]
Hi Donggyu kim,

  Unfortunately that is not the case; the software freezes during the generation of the basegrid. I managed to get through that by lowering the processor count from 16 to 8. Now I am stuck trying to add Boundary Layer Meshes; SnappyHexMesh doesn't add any layers to my geometry case. I am going to start the project from scratch and see where that goes.

-Sam

**Comment by samc99us** [comment]
To clarify the issue, this is the result I get after hitting "generate" on the base grid:
![Baram_Base_Grid_Issue](https://github.com/user-attachments/assets/1fdca896-e683-4aa8-901f-60b1b97d07c7)

**Comment by samc99us** [comment]
Some more updates:
1) I can get through the initial meshing if I don't use parallel processing. This is clearly buggy.
2) I cannot generate a boundary layer mesh. BARAM doesn't recognize my base surface as an eligible boundary for boundary layer meshing.
3) I cannot export the BaramFlow project. Baram is hung up on "export files".

For reference, I am using BaramMesh 24.2.1 on Windows 11 Pro 23H2. I have a pretty top of the line machine with 24 core processor (Intel i9-14900) and 128gb of RAM.

Any help is appreciated.

Best,
Sam

**Comment by kdg5424** [comment]
Hello Sam. This is Donggyu.
Yes. It looks like the bug. So could you give us the log file and some settings, configuration that you did?
Let us check with that.

You can see log file in the folder.
![image](https://github.com/user-attachments/assets/7e013291-37e5-4798-b7fe-53034afe4fcb)

Thanks

Donggyu

**Comment by samc99us** [comment]
Attached are the log file. The complete case file is too large to place here.

For settings, I'm essentially replicating your ahmed body example (https://baramcfd-org.translate.goog/mesh/2023/09/05/AhmedBody-post/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp), but with my own input file. For this round, I didn't attempt to simplify anything with symmtery. I did set parallel processors to 1 to get this far.

[log.txt](https://github.com/user-attachments/files/16342885/log.txt)

**Comment by kdg5424** [comment]
Hello Sam. Thank you for your log file.
If the file size is too large, could you capture your settings?
We'll check your log file and settings.

Thank you

**Comment by jiban** [comment]
Hi, Sam. This is Jake, one of the developers of BARAM.
I just want to let you know a few notes before proceeding.
1. BARAM frequently modifies the files and directories in the project during operation. If you open a windows *explorer* in one of the project directory or open a file in the project, it may cause a problem for BARAM to handle them. I'm suspecting this from your logs.  
    Please have no other application run in the project directory.
    ```python
    PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\samc9\\Dropbox\\Business\\Carter Composites\\CFD\\SharpBody2\\case\\delete_me_c8396db0-6788-46f4-bdbe-9e7405f9ded7\\constant\\polyMesh'
    [2024-07-22 16:59:06,427][qasync._QEventLoop] ==> Task exception was never retrieved
    future: <Task finished name='Task-2720' coro=<to_thread() done, defined at C:\Users\jakej\AppData\Local\Programs\Python\Python39\lib\asyncio\threads.py:12> exception=PermissionError(13, 'The process cannot access the file because it is being used by another process')>
    ```  
1. *hex6* is usually used for *farfield* boundary, and its surfaces do not support boundary layer addition in BARAM for now.

Could you please try the case again with these in mind?

Best regards,
Jake

**Comment by samc99us** [comment]
Hi Jake:

1. I assumed that might be the case; I moved the project file out of a dropbox directory to reduce/eliminate this issue.
2. I understand, I a not trying to add boundary layers to the farfield boundary. Rather, I am trying to add them to my 'ahmed body', i.e  body of revolution, or .STL file. This does not appear to be functional. See the below screenshot:
![Screenshot 2024-07-23 204003](https://github.com/user-attachments/assets/aca89b94-5131-42de-93bf-38d0cdff8575)
You can see that this is setup as a Boundary on the geometry tab, but does not continue to move through the meshing process as such:
![image](https://github.com/user-attachments/assets/8b2a31e4-1d4f-4550-9c51-aca670265c8c)

More importantly, when performing the "base grid", i.e calling blockmesh, a geometry and boundary file are generated for all the other surfaces except this. Interestingly, working with the ahmedbody.stl, I do get a boundary file upon running the base grid command. I cannot say what the difference in these two file is.

This case file might load:
[SharpBodySurface.zip](https://github.com/user-attachments/files/16354181/SharpBodySurface.zip)

**Comment by jiban** [comment]
Dear Sam,
Thank you for sharing the case file.
I think *refinement1* is just for refinement.
If so, it should not be configured as a *boundary*.

![image](https://github.com/user-attachments/assets/bdbcbb26-608e-41d5-abd4-12d7f1c52a43)

The *boundary* prevents  *snappyHexMesh* from reaching the internal space of it.
(This is the way *snappyHexMesh* is working)


I think that the following result might be what you want to have.

![image](https://github.com/user-attachments/assets/c00cf7cf-ac9a-419a-9c8b-21d53eb21790)

And following is the *geometry* configuration for it.
(I just changed the type of *refinement1_surface* to *None*)
![image](https://github.com/user-attachments/assets/4f6dc00a-abec-401f-82eb-4f05864b1956)

Thank you.

Best regards,
Jake

**Comment by samc99us** [comment]
Hi Jake,

  That did the trick. Now I'm trying to actually view the final mesh. Two errors:

1) Between the "Snap" and "Boundary Layer" functions, no change in cell count happens. I.e, when moving from the "Boundary Layer" function to the "Export Function", no change occurs to cell count, despite this being the case in prior steps.

2) I'm unsure how get the cutaway view you show under the "Snap" routine image above. This would be very helpful for checking mesh quality.

**Comment by samc99us** [comment]
Update: I'm pretty sure no boundary layers are being added to my mesh. The cell count doesn't change, as verified in BaramFlow. It would be nice to understand how to fully visualize the mesh in BaramMesh as well. Attached is a simplified version of the case file.
[SharpBody.zip](https://github.com/user-attachments/files/16365947/SharpBody.zip)

**Comment by samc99us** [comment]
I spent some more time on this today, I found the display function to cut the model-very handy. Unfortunately I could not add boundary layers, despite trying a number of different settings. I suspect this has to do with my model geometry and some bugs in SnappyHexMesh.

**Reply by jiban** [reply]
Dear Sam,
I've tried your case and got the same result.
Boundary layers were not added with the configuration.

I think mesh generation is a kind of dark magic, and the way *snappyHexMesh* is working is quite complicated.
Anyway, I've added surface refinement around your geometry and could get boundary layers like following.

![image](https://github.com/user-attachments/assets/dd97f7fd-cc9d-40ad-a13f-37ca2726351e)


And this is the surface refinement I've added.

![image](https://github.com/user-attachments/assets/32c7aa5c-fb33-4cce-8ffe-0c2c1ca493eb)

Thank you.

Best regards,
Jake

**Reply by samc99us** [reply]
Jake,

  Thank you for taking a look at this. The surface refinement did the trick! Hopefully this proves helpful to others that may run into this issue in the future.

Best,
Sam
